web/src/main/java/com/bazaarvoice/emodb/web/scanner/ScanUploader.java

```diff
@@ -47,18 +47,16 @@
     private final ScanWorkflow _scanWorkflow;
     private final ScanStatusDAO _scanStatusDAO;
     private final StashStateListener _stashStateListener;
-    private final DefaultCompactionControlManager _defaultCompactionControlManager;
-    private final DataCenters _dataCenters;
+    private final List<CompactionControlSource> _compactionControlSourceList;
```

Comment:
Would it make sense to pass in an implementation of `CompactionControlSource` which itself delegates to all other `CompactionControlSources`?  Here's my thinking:

Every time the list is iterated there's a chance that one of the calls will fail.  There's all sorts of failure and retry logic that goes into that, and that will have to be repeated on every use in this class.  Seems more natural to centralize that in an implementation which just mimics what you've already done here:  Iterate of the list of delegates, make the call on each, and respond to errors appropriately.